src/signals-unix.c

```diff
@@ -311,8 +311,14 @@ static int jl_is_on_sigstack(jl_ptls_t ptls, void *ptr, void *context)
             is_addr_on_sigstack(ptls, (void*)jl_get_rsp_from_ctx(context)));
 }
 
+extern volatile int ld_library_path_is_set;
+
 static void segv_handler(int sig, siginfo_t *info, void *context)
 {
+    if(ld_library_path_is_set) {
```

Comment:
This needs to be later. Just before the SIGDIE handler below. We use SIGSEGV for other things.